.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,172 @@
+---
+name: Test Linux
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+      test-rust:
+        description: >
+          Decides whether we perform rust tests
+        type: boolean
+        required: true
+      test-images:
+        description: >
+          Decides whether we perform image tests
+        type: boolean
+        required: true
+      install-optionals:
+        description: >
+          Decides whether we install optyional dependencies
+        type: boolean
+        default: false
+
+      install-from-sdist:
+        description: >
+          Decides whether we perform rust tests
+        type: boolean
+        default: false
+
+  merge_group:
+
+jobs:
+  tests-linux:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ matrix.os }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ matrix.os }}
+    env:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: ${{ github.workspace }}/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+      HAVE_VISUAL_TESTS_RUN: false
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "ubuntu-24.04-arm"]
```

Comment:
I'm fine with this, but does this mean the preliminary stage for 3.9 is run in duplicate on arm and x86? I think that's actually fine but I want to know if that was actually what's going to happen and if that's your intent.

Reply:
Runs as duplicate as of now.